engine/crates/composition/src/subgraphs/keys.rs

```diff
@@ -97,12 +139,39 @@ impl<'a> KeyWalker<'a> {
 }
 
 impl<'a> DefinitionWalker<'a> {
+    pub fn is_entity(self) -> bool {
+        self.entity_keys().next().is_some()
+            || self
+                .subgraphs
+                .keys
+                .nested_key_fields
+                .objects_with_nested_keys
+                .contains(&self.id)
+    }
+
     pub fn entity_keys(self) -> impl Iterator<Item = KeyWalker<'a>> {
-        let start = self.subgraphs.keys.0.partition_point(|(parent, _)| *parent < self.id);
-        self.subgraphs.keys.0[start..]
+        let start = self
+            .subgraphs
+            .keys
+            .keys
+            .partition_point(|(parent, _)| *parent < self.id);
+        self.subgraphs.keys.keys[start..]
```

Comment:
Again, there's a possibility for some generic function that could handle the partitioning.